src/problems/0053-maximum-subarray/Solution.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+
+namespace leetcode.problems.MaximumSubarray
+{
+    public class Solution
+    {
+        public int MaxSubArray(int[] nums)
+        {
+            var currentMax = int.MinValue;
+            var runningTotal = 0;
+
+            foreach (var current in nums)
+            {
+                if (runningTotal + current < current) runningTotal = 0;
```

Comment:
`runningTotal + current < current` is this the same as `runningTotal < 0`?

Reply:
Joking aside, originally before using `Math.Max()` in the `main` branch, we always add current to running total.  So if the current running total is negative, set it to zero, since the next line will be adding the current number to the current running total (which is zero).

But then you helped me realize that:
```
if (runningTotal + current < current) runningTotal = 0;
runningTotal += current;
currentMax = Math.Max(currentMax, runningTotal);
```

is the same as:
```
runningTotal = Math.Max(runningTotal + current, current);
currentMax = Math.Max(currentMax, runningTotal);
```